crates/jiter/src/python.rs

```diff
@@ -171,38 +210,55 @@ impl<'j, StringCache: StringMaybeCache, KeyCheck: MaybeKeyCheck, ParseNumber: Ma
         py: Python<'py>,
         peek_first: Peek,
         vec: &mut SmallVec<[Bound<'py, PyAny>; 8]>,
-    ) -> JsonResult<()> {
-        let v = self._check_take_value(py, peek_first)?;
+        mut build_path: impl MaybeBuildArrayPath,
+    ) -> PythonJsonResult<()> {
+        let v = self
```

Comment:
To avoid having to map all over the place, I wonder whether something like

```rust
BuildPath::enter_array(|b: &mut BuildPath::ArrayIndex| { /* do stuff */ })
```

would be more ergonomic, where you call `b.set_index()` to update the index, and then `enter_array` can be responsible for adding the path to errors inside the closure.

Similar thought for objects.

... but I still wonder if having some `parser.path()` method to rebuild the whole path and keep it out of the rest of the control flow is better at the cost of 2x slowdown on errors.